### ISSUE TITLE
Fix FP S4023 (`no-empty-interface`): Exclude extensions of TypeScript utility types

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#3634]
`extends` clause might amend the semantic of the basic interface. E.g.:
```ts
interface Foo extends Omit<Bar, 'a', 'b', 'd'> {}
```

Utility types: https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys

Sources: (internal link) https://discuss.sonarsource.com/t/typescript-s4023-false-positive/12815/1
